source/as-stream.js

```diff
@@ -50,13 +52,27 @@ module.exports = options => {
 			proxy.emit('error', new ReadError(error, options));
 		});
 
-		response.pipe(output);
-
 		if (options.throwHttpErrors && statusCode !== 304 && (statusCode < 200 || statusCode > 299)) {
 			proxy.emit('error', new HTTPError(statusCode, response.statusMessage, response.headers, options), null, response);
 			return;
 		}
 
+		finished = true;
+
+		response.pipe(output);
+
+		for (const destination of piped) {
+			if (!destination.headersSent) {
+				for (const [key, value] of Object.entries(response.headers)) {
+					if (key.toLowerCase() !== 'content-encoding') {
```

Comment:
Can you add a short code comment why we don't set this? Are there other headers we shouldn't overwrite?

Reply:
> Can you add a short code comment why we don't set this?

Yes.

> Are there other headers we shouldn't overwrite?

No.